mlir/include/mlir/Dialect/DLTI/DLTI.h

```diff
@@ -28,6 +28,12 @@ namespace dlti {
 /// query interface-implementing attrs, starting from attr obtained from `op`.
 FailureOr<Attribute> query(Operation *op, ArrayRef<DataLayoutEntryKey> keys,
                            bool emitError = false);
+
+/// Perform a DLTI-query at `op` using each string in `keys` as a separate DLTI
+/// entry key, recursively querying on query interface-implementing attrs,
+/// starting from attr obtained from `op`.
+FailureOr<Attribute> query(Operation *op, ArrayRef<StringRef> keys,
```

Comment:
Can we spell out the APIs in terms of StringAttr? This turns string comparisons to pointer comparisons.

Reply:
Oh you're actually just wrapping another API already taking StringAttr, and doing the conversion, forget about my comment then! :)